functions/uploads.py

```diff
@@ -253,18 +254,41 @@ def _gcs_add_prefix_reader_permission(group_email: str, prefix: str):
         .isoformat()
     )
 
-    prefixCheck = f'resource.name.startsWith("projects/_/buckets/{GOOGLE_DATA_BUCKET}/objects/{cleaned_prefix}/")'
-    expiryCheck = f'request.time < timestamp("{grant_until_date}T00:00:00Z")'
+    prefix_check = f'resource.name.startsWith("projects/_/buckets/{GOOGLE_DATA_BUCKET}/objects/{cleaned_prefix}/")'
+    expiry_check = f'request.time < timestamp("{grant_until_date}T00:00:00Z")'
+    group_member = f"group:{group_email}"
+
+    # look for a duplicate conditional binding
+    matching_binding_index = None
+    for i, binding in enumerate(policy.bindings):
+        role_matches = binding.get("role") == GOOGLE_ANALYSIS_GROUP_ROLE
+        member_matches = binding.get("members") == {group_member}
```

Comment:
comparison to a set seems a bit interesting to me, are you sure that `policy.bindings.members` would be a set? 
you're using just list on the line 287 below

Reply:
Yeah, that's a fair concern (though I think it does come back as a `set` from the GCS python client). An `in` comparison is probably the safest approach. Same could be said about the code in `cidc_api.shared.gcloud_client`.